core/jobs/batch_jobs/user_settings_profile_picture_jobs_test.py

```diff
@@ -0,0 +1,184 @@
+    'CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK'
+    '0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CM'
+    'K0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0C'
+    'MK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0'
+    'CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK'
+    '0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CM'
+    'K0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0C'
+    'MK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0'
+    'CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK'
+    '0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CMK0CM'
+    'K0CMK0CMO0TAAD//2Anhf4QtqobAAAAAElFTkSuQmCC'
+)
+
+
+class AuditInvalidProfilePictureJobTests(job_test_utils.JobTestBase):
+    """Fetch invalid profile pictures data."""
+
+    JOB_CLASS = user_settings_profile_picture_jobs.AuditInvalidProfilePictureJob
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.user_1 = self.create_model(
+            user_models.UserSettingsModel,
+            id='test_id_1',
+            email='test_1@example.com',
+            username='test_1',
+            roles=[feconf.ROLE_ID_FULL_USER, feconf.ROLE_ID_CURRICULUM_ADMIN],
+            profile_picture_data_url=BASE64_STR
+        )
+
+        self.user_2 = self.create_model(
+            user_models.UserSettingsModel,
+            id='test_id_2',
+            email='test_2@example.com',
+            username='test_2',
+            roles=[feconf.ROLE_ID_FULL_USER, feconf.ROLE_ID_CURRICULUM_ADMIN],
+        )
+
+    def test_run_with_no_models(self) -> None:
+        self.assert_job_output_is([])
+
+    def test_get_invalid_image_dimension_data(self) -> None:
+        self.put_multi([self.user_1])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout='TOTAL INVALID IMAGES SUCCESS: 1'
+            ),
+            job_run_result.JobRunResult(
+                stderr=(
+                    'The username is test_1 and the invalid image details '
+                    'are [\'wrong dimensions - height = 256 and width = 256\'].'
+                )
+            )
+        ])
+
+    def test_get_invalid_image_base64_data(self) -> None:
+        self.put_multi([self.user_2])
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout='TOTAL INVALID IMAGES SUCCESS: 1'
+            ),
+            job_run_result.JobRunResult(
+                stderr=(
+                    'The username is test_2 and the invalid image details '
+                    'are [\'Image is not base64 having value - None\'].'
+                )
+            )
+        ])
+
+    def test_ignore_valid_images(self) -> None:
+        self.user_1.profile_picture_data_url = (
+            user_services.DEFAULT_IDENTICON_DATA_URL)
+        self.put_multi([self.user_1])
+        self.assert_job_output_is([])
```

Comment:
Hey @vojtechjelinek,
Please take a look at this test, Here I have assigned the `DEFAULT_IDENTICON_DATA_URL` as the profile picture URL which is for the `profile_picture_data_url`. Now as this is the test for the audit job I should not get any output because we have a valid `profile_picture_data_url` but it fails. So, I found out that I'm not able to calculate the size of the image in the audit job where I'm calculating the height and width. Also when I checked online to convert this default URL to image I was getting error. I also tried removing the `data:image/png;base64,` in front of the default URL but the image was still not loading. You can take a look at the Audit job having `_get_invalid_image` function which I use to filter the image. Please note that I also tried removing the line `63` from the function in order to check but still the audit job was not successful.
Please note that the `BASE64_STR` variable I define is for the base64 string that is successfully getting converted to image and the tests are passing.
I will link the details to the error once the tests run completely.
Thanks

Reply:
So the default image is an exception from our normal images which are 150 × 150, right?

Reply:
Yes, and at all the places where we have default images, they all have the wrong dimensions.

Reply:
Reporting the default images.